charts/gsp-cluster/templates/01-aws-system/aws-service-operator.yaml

```diff
@@ -0,0 +1,232 @@
+    version: v1alpha1
+
+- apiVersion: apiextensions.k8s.io/v1beta1
+  kind: CustomResourceDefinition
+  metadata:
+    name: s3buckets.service-operator.aws
+  spec:
+    group: service-operator.aws
+    names:
+      kind: S3Bucket
+      listKind: S3BucketList
+      plural: s3buckets
+      shortNames:
+      - s3
+      - bucket
+      - buckets
+      singular: s3bucket
+    scope: Namespaced
+    version: v1alpha1
+
+- apiVersion: apiextensions.k8s.io/v1beta1
+  kind: CustomResourceDefinition
+  metadata:
+    name: snssubscriptions.service-operator.aws
+  spec:
+    group: service-operator.aws
+    names:
+      kind: SNSSubscription
+      listKind: SNSSubscriptionList
+      plural: snssubscriptions
+      shortNames:
+      - subscription
+      singular: snssubscription
+    scope: Namespaced
+    version: v1alpha1
+
+- apiVersion: apiextensions.k8s.io/v1beta1
+  kind: CustomResourceDefinition
+  metadata:
+    name: snstopics.service-operator.aws
+  spec:
+    group: service-operator.aws
+    names:
+      kind: SNSTopic
+      listKind: SNSTopicList
+      plural: snstopics
+      shortNames:
+      - sns
+      - topic
+      - topics
+      singular: snstopic
+    scope: Namespaced
+    version: v1alpha1
+
+- apiVersion: apiextensions.k8s.io/v1beta1
+  kind: CustomResourceDefinition
+  metadata:
+    name: sqsqueues.service-operator.aws
+  spec:
+    group: service-operator.aws
+    names:
+      kind: SQSQueue
+      listKind: SQSQueueList
+      plural: sqsqueues
+      shortNames:
+      - sqs
+      - queue
+      - queues
+      singular: sqsqueue
+    scope: Namespaced
+    version: v1alpha1
+
+- kind: ClusterRole
+  apiVersion: rbac.authorization.k8s.io/v1beta1
+  metadata:
+    name: aws-service-operator
+  rules:
+  - apiGroups:
+    - ""
+    resources:
+    - secrets
+    - pods
+    - configmaps
+    - services
+    - events
+    verbs:
+    - get
+    - list
+    - watch
+    - create
+    - delete
+    - update
+    - patch
+  - apiGroups:
+    - apiextensions.k8s.io
+    resources:
+    - customresourcedefinitions
+    verbs:
+    - get
+    - list
+    - watch
+    - create
+    - delete
+  - apiGroups:
+    - service-operator.aws
+    resources:
+    - "*"
+    verbs:
+    - "*"
+
+- kind: ServiceAccount
+  apiVersion: v1
+  metadata:
+    name: aws-service-operator
+    namespace: aws-service-operator
+
+- kind: ClusterRoleBinding
+  apiVersion: rbac.authorization.k8s.io/v1beta1
+  metadata:
+    name: aws-service-operator
+  roleRef:
+    apiGroup: rbac.authorization.k8s.io
+    kind: ClusterRole
+    name: aws-service-operator
+  subjects:
+  - kind: ServiceAccount
+    name: aws-service-operator
+    namespace: aws-service-operator
+
+- kind: Deployment
+  apiVersion: apps/v1beta1
+  metadata:
+    name: aws-service-operator
+    namespace: aws-service-operator
+  spec:
+    replicas: 1
+    template:
+      metadata:
+        annotations:
+          iam.amazonaws.com/role: aws-service-operator
```

Comment:
Come the thought of it I'm not sure what this is used for, as when I was testing it I provided it some credentials directly, and when running on AWS it would presumably just use the credentials provided to it. I don't think it would need to assume a particular role.

Reply:
Nope, it does not exist where I was testing.

Reply:
#259